src/ext/java/org/opentripplanner/ext/emissions/Co2EmissionsDataReader.java

```diff
@@ -38,20 +39,23 @@ public Co2EmissionsDataReader(DataImportIssueStore issueStore) {
    */
   public Map<FeedScopedId, Double> readGtfs(File directory) {
     String feedId = "";
-    Map<FeedScopedId, Double> emissionsData = new HashMap<>();
-    try (InputStream feedInfoStream = new FileInputStream(directory + "/feed_info.txt")) {
-      feedId = readFeedId(feedInfoStream);
-    } catch (IOException e) {
-      issueStore.add("InvalidEmissionData", "Reading feed_info.txt failed.");
-      LOG.error("InvalidEmissionData: reading feed_info.txt failed.", e);
-    }
-    try (InputStream stream = new FileInputStream(directory + "/emissions.txt")) {
-      emissionsData = readEmissions(stream, feedId);
-    } catch (IOException e) {
-      issueStore.add("InvalidEmissionData", "Reading emissions.txt failed.");
-      LOG.error("InvalidEmissionData: reading emissions.txt failed.", e);
+    File feedFile = new File(directory + "/feed_info.txt");
+    File emissionsFile = new File(directory + "/emissions.txt");
+    if (feedFile.exists() && emissionsFile.exists()) {
+      try (InputStream feedInfoStream = new FileInputStream(feedFile)) {
+        feedId = readFeedId(feedInfoStream);
+      } catch (IOException e) {
+        issueStore.add("InvalidEmissionData", "Reading feed_info.txt failed.");
+        LOG.error("InvalidEmissionData: reading feed_info.txt failed.", e);
+      }
+      try (InputStream stream = new FileInputStream(emissionsFile)) {
+        return readEmissions(stream, feedId);
+      } catch (IOException e) {
+        issueStore.add("InvalidEmissionData", "Reading emissions.txt failed.");
+        LOG.error("InvalidEmissionData: reading emissions.txt failed.", e);
+      }
     }
-    return emissionsData;
+    return null;
```

Comment:
If you return an empty Map (`Map.of()`) you don't need a null check at the call site.